src/sentry/web/frontend/transfer_project.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+
+import six
+from uuid import uuid4
+from six.moves.urllib.parse import urlencode
+
+from django import forms
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from sentry import roles, options
+from sentry.web.frontend.base import ProjectView
+from sentry.utils.email import MessageBuilder
+from sentry.utils.signing import sign
+from sentry.models import AuditLogEntryEvent, OrganizationMember, User
+
+
+class TransferProjectForm(forms.Form):
+    email = forms.CharField(
+        label=_('Organization Owner'),
+        max_length=200,
+        widget=forms.TextInput(attrs={'placeholder': _('user@company.com')})
+    )
+
+
+class TransferProjectView(ProjectView):
+    required_scope = 'project:admin'
+    sudo_required = True
+
+    def get_form(self, request):
+        if request.method == 'POST':
+            return TransferProjectForm(request.POST)
+        return TransferProjectForm()
+
+    def handle(self, request, organization, team, project):
+        form = self.get_form(request)
+
+        if form.is_valid():
+            email = form.cleaned_data.get('email')
+            try:
+                owner = User.objects.get(email__iexact=email, is_active=True)
+            except User.DoesNotExist:
+                messages.add_message(
```

Comment:
I think this is a bit misleading. You're strictly checking if _any_ user exists here, and if there is a user, you then proceed to check if they are an owner. But if they're not an owner, you don't error at all.

Why not just combine the `OrganizationMember` query all in 1?

```python
try:
    owner = OrganizationMember.objects.filter(
        role=roles.get_top_dog().id,
        user__email__iexact=email,
        user__is_active=True,
    )
except OrganizationMember.DoesNotExist:
    # handle this
```

Then you can access `owner.user_id` to get the user id.

Reply:
was going off this comment lol https://github.com/getsentry/sentry/commit/582df349124344d4a2579d161ba691eb4fe402f5#commitcomment-24415997

Reply:
From what I can tell, since it's a bit hard to view all the state changes, the logic was slightly different than this.

I was proposing what I did to reduce the queries being performed because it was redundant. But now as-is, we can reduce it even more. So I might have been a bit too quick on suggesting an actual improvement, but my point was that there was unnecessary work that was being done.

What I most recently proposed seems to cut it down to the bare minimum for what we need since you don't actually need the `User` object anymore anywhere in here. Which means we can remove the `User` query and the `exists()` query.

Reply:
yeah agreed. 
```python
owner = OrganizationMember.objects.filter(
        role=roles.get_top_dog().id,
        user__email__iexact=email,
        user__is_active=True,
    )
```
would return owner as a query set though right? what would be the best way to get the owner id?

Reply:
Damn, another mistype. You want `.get()` instead of `.filter()` there.

```python
owner = OrganizationMember.objects.get(
    role=roles.get_top_dog().id,
    user__email__iexact=email,
    user__is_active=True,
)
```

Because you're right, it would have been a QuerySet.

Now with an `owner` which is a `OrganizationMember` object, you can access `owner.user_id` to get the user id of the owner.

Reply:
👍